identity-account/src/types/method_content.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_core::crypto::KeyType;
+use identity_core::crypto::PrivateKey;
+use identity_core::crypto::PublicKey;
+use identity_did::verification::MethodType;
+
+/// Method content for creating new verification methods.
+#[derive(Clone, Debug)]
+pub enum MethodContent {
```

Comment:
Can you remind me what the problem with this was (as an effort to reduce how fast the enum needs to grow with future key types):

```rust
pub enum MethodContent {
  Generate(KeyType),
  PrivateKey(KeyType, PrivateKey),
  PublicKey(KeyType, PublicKey),
}
```

As it is now, we also don't have a guarantee that e.g. `PrivateEd25519(PrivateKey)` actually contains the correct number of bytes for an ed25519 key, so I'm wondering if we actually lose any validation with the above approach.

If we keep it as-is, shouldn't it be `#[non_exhaustive]`?

Reply:
It might be possible to go with a simpler enum if we refactor `PrivateKey` and `PublicKey` to be traits with concrete implementations for `X25519PublicKey`, `Ed25519PrivateKey` etc.

Will keep the current naming scheme.